src/lib.rs

```diff
@@ -36,6 +36,63 @@ pub use types::geometrycollection::GeometryCollection;
 
 pub use types::feature::Feature;
 pub use types::featurecollection::FeatureCollection;
-pub use types::featurecollection::from_str;
+
+macro_rules! expect_string {
+    ($value:expr) => (try!(
+        match $value.as_string() {
```

Comment:
Depending on whether or not you want to use unstable features, you could utilize [`ok_or`](http://doc.rust-lang.org/std/option/enum.Option.html#method.ok_or) here